include/swift/AST/DiagnosticsParse.def

```diff
@@ -990,7 +990,9 @@ ERROR(expected_dollar_numeric,none,
 ERROR(dollar_numeric_too_large,none,
       "numeric value following '$' is too large", ())
 ERROR(numeric_literal_numeric_member,none,
-   "expected named member of numeric literal", ())
+      "expected named member of numeric literal", ())
+NOTE(suggest_backtick_dollar_identifier,none,
+     "to use '$' as an identifier use backticks to escape it", ())
```

Comment:
I think you should fold the error and note together, and put the Fix-It on the error itself. Perhaps wording of something like:

  '$' is not an identifier; use backticks (`$`) to escape it